wm/frame.go

```diff
@@ -3,271 +3,214 @@
-		f.height = f.height + uint16(deltaY)
-		f.mouseX = x
-		f.mouseY = y
+	if fr.mouseResize {
+		fr.width = fr.width + uint16(deltaX)
+		fr.height = fr.height + uint16(deltaY)
+		fr.mouseX = x
+		fr.mouseY = y
 
-		borderWidth := 2 * f.wm.borderWidth()
-		if f.width < uint16(f.minWidth)+borderWidth {
-			f.width = uint16(f.minWidth) + borderWidth
-		}
-		borderHeight := 2*f.wm.borderWidth() + f.wm.titleHeight()
-		if f.height < uint16(f.minHeight)+borderHeight {
-			f.height = uint16(f.minHeight) + borderHeight
+		borderWidth := 2 * fr.cli.wm.borderWidth()
+		if fr.width < uint16(fr.minWidth)+borderWidth {
+			fr.width = uint16(fr.minWidth) + borderWidth
 		}
-
-		if f.framed {
-			err := xproto.ConfigureWindowChecked(f.wm.x.Conn(), f.win, xproto.ConfigWindowX|xproto.ConfigWindowY|
-				xproto.ConfigWindowWidth|xproto.ConfigWindowHeight,
-				[]uint32{uint32(f.wm.borderWidth()), uint32(f.wm.borderWidth() + f.wm.titleHeight()),
-					uint32(f.width - borderWidth), uint32(f.height - borderHeight)}).Check()
-			if err != nil {
-				log.Println("ConfigureWindow Err", err)
-			}
-
-			f.ApplyTheme()
+		borderHeight := 2*fr.cli.wm.borderWidth() + fr.cli.wm.titleHeight()
+		if fr.height < uint16(fr.minHeight)+borderHeight {
+			fr.height = uint16(fr.minHeight) + borderHeight
 		}
+		moveOnly = false
 	} else {
-		f.x += deltaX
-		f.y += deltaY
-	}
-
-	err := xproto.ConfigureWindowChecked(f.wm.x.Conn(), f.id, xproto.ConfigWindowX|xproto.ConfigWindowY|
-		xproto.ConfigWindowWidth|xproto.ConfigWindowHeight,
-		[]uint32{uint32(f.x), uint32(f.y), uint32(f.width), uint32(f.height)}).Check()
-	if err != nil {
-		log.Println("ConfigureWindow Err", err)
+		fr.x += deltaX
+		fr.y += deltaY
 	}
+	fr.updateGeometry(uint16(fr.x), uint16(fr.y), fr.width, fr.height, moveOnly)
 }
 
-func (f *frame) motion(x, y int16) {
+func (fr *frame) motion(x, y int16) {
 	cursor := defaultCursor
-	if x > int16(f.wm.borderWidth()) && x <= int16(f.wm.borderWidth()+f.wm.buttonWidth()) &&
-		y <= int16(f.wm.borderWidth()+f.wm.titleHeight()) {
+	if x > int16(fr.cli.wm.borderWidth()) && x <= int16(fr.cli.wm.borderWidth()+fr.cli.wm.buttonWidth()) &&
+		y <= int16(fr.cli.wm.borderWidth()+fr.cli.wm.titleHeight()) {
 		cursor = closeCursor
-	} else if x >= int16(f.width-f.wm.buttonWidth()) && y >= int16(f.height-f.wm.buttonWidth()) {
+	} else if x >= int16(fr.width-fr.cli.wm.buttonWidth()) && y >= int16(fr.height-fr.cli.wm.buttonWidth()) {
 		cursor = resizeCursor
 	}
 
-	err := xproto.ChangeWindowAttributesChecked(f.wm.x.Conn(), f.id, xproto.CwCursor,
+	err := xproto.ChangeWindowAttributesChecked(fr.cli.wm.x.Conn(), fr.cli.id, xproto.CwCursor,
 		[]uint32{uint32(cursor)}).Check()
 	if err != nil {
-		log.Println("SetCursor Err", err)
+		fyne.LogError("Set Cursor Error", err)
 	}
 }
 
-func (f *frame) RaiseAbove(win desktop.Window) {
-	topID := f.wm.rootID
-	if win != nil {
-		topID = win.(*frame).id
+func (fr *frame) updateGeometry(x uint16, y uint16, w uint16, h uint16, moveOnly bool) {
```

Comment:
This function is always passed the frame x, y, width, height. Either this function should update the values - or it should not take the additional parameters...